### PR TITLE
fix(Core/Pet): Risen Ghoul random name & pet detail

### PR DESCRIPTION
**Pull request by TheWatchingI**
## CHANGES PROPOSED:
-  Shows an unique name & pet details for a permanent "Risen Ghoul" created with the spell "Raise Dead" and the unholy talent "Master of Ghouls" (which makes the pet permanent)

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2942 while still preserving the changes concerning the bug with permanent water elementals in https://github.com/azerothcore/azerothcore-wotlk/pull/2637

## TESTS PERFORMED:
- tested build on Windows 10
- tested successfully in-game (works on permanent ghouls, ignores temporary ones or those summoned by the spell "Army Of The Dead" and the pet details of permanent Water Elementals are still hidden)

## HOW TO TEST THE CHANGES:
- play death knight
- activate the talent "Master of Ghouls" in the unholy tree
- Learn & use the spell "Raise Dead"

If you want to check, if the pet details of the permanent Water Elemental are still hidden
- play mage
- insert major glyph "Glyph of Eternal Water" (id: 50045)
- learn spell "Summon Water Elemental" (id: 31687) and use it

## KNOWN ISSUES AND TODO LIST:
none

## Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Handlers/PetHandler.cpp [hunk 1]
```diff
@@ -160,11 +160,11 @@ uint8 WorldSession::HandleLoadPetFromDBFirstCallback(PreparedQueryResult result,
...
     }
 
-    if (pet->getPetType() == HUNTER_PET || pet->GetCreatureTemplate()->type == CREATURE_TYPE_DEMON)
-        pet->GetCharmInfo()->SetPetNumber(pet_number, pet->IsPermanentPetFor(owner)); // Show pet details tab (Shift+P) only for hunter pets or demons
+    if (pet->getPetType() == HUNTER_PET || pet->GetCreatureTemplate()->type == CREATURE_TYPE_DEMON || pet->GetCreatureTemplate()->type == CREATURE_TYPE_UNDEAD)
+        pet->GetCharmInfo()->SetPetNumber(pet_number, pet->IsPermanentPetFor(owner)); // Show pet details tab (Shift+P) only for hunter pets, demons or undead
     else
         pet->GetCharmInfo()->SetPetNumber(pet_number, false);
-
+
```
Comment (Kitzunu) [inline src/server/game/Handlers/PetHandler.cpp:167]:
```suggestion

```

**Comment by Kitzunu** [comment]
Should you really be able to see pet details for DK pets?
From the previous PR
>pet details should not show for the Death Knight's permanent ghoul or the Mage's permanent water elemental

**Comment by TheWatchingI** [comment]
> 
> 
> Should you really be able to see pet details for DK pets?
> From the previous PR
> 
> > pet details should not show for the Death Knight's permanent ghoul or the Mage's permanent water elemental
> 
> I think it should only be random name

If I understand this here correctly https://www.engadget.com/2008-09-10-death-knight-ghouls-baby-name-book.html the random name is a post-Wotlk addition.

Concering the previous PR: The problem was that the water elemtal has a visual bug on its pet detail page. The ghoul doesn't have that - in fact, it works absolutely perfect (not only the picture, but also the display of level and race).

WoWScrnShot_042620_153510
: 

I'm not 100% sure, if the pet details were visible on the official servers, but if I understand the previous PR correctly, the details are connected to displaying the unique names (which was in fact a wotlk feature (For example see this video of wotlk 3.3.5 https://youtu.be/Cp3nIhf2tfk?t=210)

**Comment by wizzymore** [comment]
Based on this video:
https://www.youtube.com/watch?v=c4mIHZHd050

The pet tab was there for DK as well.

**Comment by TheWatchingI** [comment]
The existence of the pet tab isn't enough, because it also holds all mounts. One would need a video or a screenshot with the pet tab open & a permanent (!) ghoul currently active to see, if the option is actually there.
The closest I could find is a post of Mist of Pandaria with the picture of a ghoul with ghost skin activated http://ctrlaltwow.blogspot.com/2014/02/ctrl-alt-wow-episode-355-knock-knock.html

That isn't an absolute proof, but I would say that it makes it much more likely that Wotlk hat the stat page too.

**Comment by Kitzunu** [comment]
Okay I did some reading up and I am okay with these changes as long as:
1. Non-permanent ghouls (the ghouls of DK's that has not specced Unholy with [Master of Ghouls](https://wowwiki.fandom.com/wiki/Master_of_Ghouls) do not show on the pet tab. This includes Raise Ally & Army of the Dead even if the DK is specced with Master of Ghouls.
2. These changes should NOT rename the "Risen Ally" ghoul nor should it rename the ghouls of Army of the Dead

**Comment by TheWatchingI** [comment]
Since I'm new to this: Can I test my own stuff?
Probably can't hurt either way.

Here a screenshot from Army of the Dead.
WoWScrnShot_042820_151022

And here one with Risen Ally:
WoWScrnShot_042820_151303

As one can see, the pet tab only includes the mounts (no pet details) and the spawned ghouls have their usual names (and of course I still have the unholy spec).

**Comment by Kitzunu** [comment]
Of course, you're allowed to test your own stuff, however, someone else must test it too to confirm it as well :)

**Comment by BarbzYHOOL** [comment]
from the tests done, it looks very good and logic

**Comment by FrancescoBorzi** [comment]
> Of course, you're allowed to test your own stuff

you are not only "allowed", it's implicit that the author of the PR tested its own code before pushing it (unless it's specified that the code was not tested for some reasons)

**Comment by Kitzunu** [comment]
Looks bueno to me :)

**Comment by Kitzunu** [comment]
Ty for the PR!

**Pull request merged**